### DISCUSSION TITLE
Web socket support

### DISCUSSION THREAD

**Discussion by radudalbea** [adonisjs/core#2845] [General]
Guys I couldn't find a channel where to ask a question so I leave this here: When will you have socket support?
At the moment there is a guide on how to integrate socket.io but is very small. For example I need my users to be authenticated when they connect to the socket and I don't know how to do it.

Thanks in advance,
Radu

**Comment by artem-a** [comment]
Hi @radudalbea here is an example how you can have socket support [Example](https://github.com/artem-a/adonis-api-ws-server)

I hope it helps

**Reply by radudalbea** [reply]
Hi @artem-a. Thanks a lot for the example. It really helped me. I was going to do the same thing but your structure cleared a lot of boilerplates for me.

Hopefully in the future updates, Adonis will have support for sockets like the previous version. But for now I think I can live with this implementation. 

If anyone will see this topic and will need to build an in-app notification module ping me and I will tell you how I implemented it. I would post a repo link here but my project is private.
